### ISSUE TITLE
[Py3.6][Windows][v1.1.0] Fail to install through pip

### ISSUE THREAD

**Issue by ducdetronquito** [mjs/imapclient#325]
Hi :)

I have trouble to install the latest version (v1.1.0) of ìmapclient` for Python 3.6 on windows. Here is the ouput:

```bash
Collecting IMAPClient>=1.1.0 (from -r common.txt (line 84))
  Downloading IMAPClient-1.1.0.tar.gz (192kB)
    100% |████████████████████████████████| 194kB 4.8MB/s
Exception:
Traceback (most recent call last):
  File "c:\users\gpaulet\.env\migpy3\lib\site-packages\pip\compat\__init__.py", line 73, in console_to_str
    return s.decode(sys.__stdout__.encoding)
UnicodeDecodeError: 'utf-8' codec can't decode byte 0xe9 in position 5: invalid continuation byte

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "c:\users\gpaulet\.env\migpy3\lib\site-packages\pip\basecommand.py", line 215, in main
    status = self.run(options, args)
  File "c:\users\gpaulet\.env\migpy3\lib\site-packages\pip\commands\install.py", line 324, in run
    requirement_set.prepare_files(finder)
  File "c:\users\gpaulet\.env\migpy3\lib\site-packages\pip\req\req_set.py", line 380, in prepare_files
    ignore_dependencies=self.ignore_dependencies))
  File "c:\users\gpaulet\.env\migpy3\lib\site-packages\pip\req\req_set.py", line 634, in _prepare_file
    abstract_dist.prep_for_dist()
  File "c:\users\gpaulet\.env\migpy3\lib\site-packages\pip\req\req_set.py", line 129, in prep_for_dist
    self.req_to_install.run_egg_info()
  File "c:\users\gpaulet\.env\migpy3\lib\site-packages\pip\req\req_install.py", line 439, in run_egg_info
    command_desc='python setup.py egg_info')
  File "c:\users\gpaulet\.env\migpy3\lib\site-packages\pip\utils\__init__.py", line 676, in call_subprocess
    line = console_to_str(proc.stdout.readline())
  File "c:\users\gpaulet\.env\migpy3\lib\site-packages\pip\compat\__init__.py", line 75, in console_to_str
    return s.decode('utf_8')
UnicodeDecodeError: 'utf-8' codec can't decode byte 0xe9 in position 5: invalid continuation byte
```

Surprisingly, I have no issue at all when installing the 1.0.2 version.

Do you have any idea what is happening ? Does it happends to you too ?

Thanks in advance :)

**Comment by NicolasLM** [comment]
Hey, interesting enough the exception happens in pip. Do you have an updated pip? Can you try to `pip install -U pip setuptools`?

**Comment by ducdetronquito** [comment]
Hi @NicolasLM ,

`setuptools` wasn't up to date, but now my virtualenv is:
```bash
pip (9.0.1)
setuptools (38.2.4)
wheel (0.30.0)
```

Now I have one more error:

```bash
Collecting imapclient
  Downloading IMAPClient-1.1.0.tar.gz (192kB)
    100% |████████████████████████████████| 194kB 3.5MB/s
Exception:
Traceback (most recent call last):
  File "c:\users\gpaulet\.env\migpy3\lib\site-packages\pip\compat\__init__.py", line 73, in console_to_str
    return s.decode(sys.__stdout__.encoding)
UnicodeDecodeError: 'utf-8' codec can't decode byte 0xe9 in position 5: invalid continuation byte

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "c:\users\gpaulet\.env\migpy3\lib\site-packages\pip\commands\install.py", line 324, in run
    requirement_set.prepare_files(finder)
  File "c:\users\gpaulet\.env\migpy3\lib\site-packages\pip\req\req_set.py", line 380, in prepare_files
    ignore_dependencies=self.ignore_dependencies))
  File "c:\users\gpaulet\.env\migpy3\lib\site-packages\pip\req\req_set.py", line 634, in _prepare_file
    abstract_dist.prep_for_dist()
  File "c:\users\gpaulet\.env\migpy3\lib\site-packages\pip\req\req_set.py", line 129, in prep_for_dist
    self.req_to_install.run_egg_info()
  File "c:\users\gpaulet\.env\migpy3\lib\site-packages\pip\req\req_install.py", line 439, in run_egg_info
    command_desc='python setup.py egg_info')
  File "c:\users\gpaulet\.env\migpy3\lib\site-packages\pip\utils\__init__.py", line 676, in call_subprocess
    line = console_to_str(proc.stdout.readline())
  File "c:\users\gpaulet\.env\migpy3\lib\site-packages\pip\compat\__init__.py", line 75, in console_to_str
    return s.decode('utf_8')
UnicodeDecodeError: 'utf-8' codec can't decode byte 0xe9 in position 5: invalid continuation byte

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "c:\users\gpaulet\.env\migpy3\lib\site-packages\pip\basecommand.py", line 215, in main
    status = self.run(options, args)
  File "c:\users\gpaulet\.env\migpy3\lib\site-packages\pip\commands\install.py", line 385, in run
    requirement_set.cleanup_files()
  File "c:\users\gpaulet\.env\migpy3\lib\site-packages\pip\utils\build.py", line 38, in __exit__
    self.cleanup()
  File "c:\users\gpaulet\.env\migpy3\lib\site-packages\pip\utils\build.py", line 42, in cleanup
    rmtree(self.name)
  File "c:\users\gpaulet\.env\migpy3\lib\site-packages\pip\_vendor\retrying.py", line 49, in wrapped_f
    return Retrying(*dargs, **dkw).call(f, *args, **kw)
  File "c:\users\gpaulet\.env\migpy3\lib\site-packages\pip\_vendor\retrying.py", line 212, in call
    raise attempt.get()
  File "c:\users\gpaulet\.env\migpy3\lib\site-packages\pip\_vendor\retrying.py", line 247, in get
    six.reraise(self.value[0], self.value[1], self.value[2])
  File "c:\users\gpaulet\.env\migpy3\lib\site-packages\pip\_vendor\six.py", line 686, in reraise
    raise value
  File "c:\users\gpaulet\.env\migpy3\lib\site-packages\pip\_vendor\retrying.py", line 200, in call
    attempt = Attempt(fn(*args, **kwargs), attempt_number, False)
  File "c:\users\gpaulet\.env\migpy3\lib\site-packages\pip\utils\__init__.py", line 102, in rmtree
    onerror=rmtree_errorhandler)
  File "c:\users\gpaulet\.env\migpy3\lib\shutil.py", line 494, in rmtree
    return _rmtree_unsafe(path, onerror)
  File "c:\users\gpaulet\.env\migpy3\lib\shutil.py", line 384, in _rmtree_unsafe
    _rmtree_unsafe(fullname, onerror)
  File "c:\users\gpaulet\.env\migpy3\lib\shutil.py", line 384, in _rmtree_unsafe
    _rmtree_unsafe(fullname, onerror)
  File "c:\users\gpaulet\.env\migpy3\lib\shutil.py", line 393, in _rmtree_unsafe
    onerror(os.rmdir, path, sys.exc_info())
  File "c:\users\gpaulet\.env\migpy3\lib\site-packages\pip\utils\__init__.py", line 114, in rmtree_errorhandler
    func(path)
OSError: [WinError 145] Le répertoire n’est pas vide: 'C:\\Users\\gpaulet\\AppData\\Local\\Temp\\pip-build-1a3pr6z6\\imapclient\\.eggs'
```

**Comment by NicolasLM** [comment]
I think that it's a bug of pip on Windows, see https://github.com/pypa/pip/issues/4110 for a suggested workaround.

**Comment by ducdetronquito** [comment]
You were right, that's a problem with pip on windows : it is now fixed by upgrading pip from 9.0.1 to the master version.

Everything is fine now :)

Sorry for this issue, and have a nice day !

**Comment by NicolasLM** [comment]
No problem, enjoy your freshly updated IMAPClient :)
